astropy/tests/helper.py

```diff
@@ -258,7 +258,7 @@ def treat_deprecations_as_exceptions():
     # Now, start over again with the warning filters
     warnings.resetwarnings()
     # Now, turn DeprecationWarnings into exceptions
-    _all_warns = [DeprecationWarning]
+    _all_warns = [DeprecationWarning, FutureWarning]
```

Comment:
This is not backward compatible but is the laziest solution I can think of. We can be all fancy and make it configurable and such, but that would also make the code even more complicated than it already is. Also not sure how to test this.

Do we also want to change this a few lines below?

```python
warnings.filterwarnings("ignore", s, DeprecationWarning)
```

Reply:
I like the simplicity! Below, I'm not quite sure why we specifically check `DeprecationWarning` in the first place, since the filter is already done for a specific text.  So, maybe just make that category the more general `Warning`, so that it captures any type of warning that has that relevant text?

Reply:
Looks like #6223 might explain why but now the question is do we want to keep it that way?

Reply:
I think we'd keep exactly the same behaviour if we would change `DeprecationWarning` to `Warning` for that part, except that one can now also filter out `FutureWarning` (or any warning) from subpackages.

Reply:
@mhvk , is that backward compatible all the way to LTS? Or just for Python 3.5+ or ...? I haven't been keeping up with how Python refactored their warnings.

Reply:
Would certainly seem fine not to backport this to 2.0... (though I think it would in fact be fine)

Reply:
Generally I like to backport these kind of changes to LTS even when there is a slight API change, so affiliated packages can start using it, however here it's probably enough if they get the warning when using latest astropy and fix their deprecated usage based on that.

Reply:
I'll have to dig out the Python doc on warning behaviors across different Python versions but I don't have time right now. Feel free to take over this PR if you can't wait. Thanks.